### PR TITLE
Add `ButtonStyle.premium` and deprecate `InteractionCallbackType.premiumRequired`

### PR DESCRIPTION
**Pull request by MCausc78**
# Description

Implements https://github.com/discord/discord-api-docs/pull/6875

## Connected issues & other potential problems

None

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Ran `dart analyze` or `make analyze` and fixed all issues
- [x] Ran `dart format --set-exit-if-changed -l 160 ./lib` or `make format` and fixed all issues
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by abitofevrything — APPROVED** [review]

**Comment by abitofevrything** [comment]
LGTM, will wait to see if the PR on Discord's end changes in any way before merging.

**Review by l7ssha — APPROVED** [review]

**Pull request merged**
